# Esercizio 1

Siano dati i seguenti nominativi di donatori di sangue ed i corrispondenti gruppi sanguigni (GS):

In [2]:
totnomi <- c("Luca","Mario", "Tommaso", "Marco", "Gianni", "Pino", "Alessia", "Francesca", "Anna", "Stefania", "Viola", "Cristina")
totcognomi <- c("Rossi", "Neri", "Bianchi", "Verdi", "Brambilla", "Ferrari", "Russo", "Esposito", "Romani", "Marino", "Colombo", "Conti", "Gallo", "Greco")
totgruppi <- c("0", "A", "B", "AB")

Estrarre dai vettori 100 campioni ed inserire nomi, cognomi e GS in un dataframe con colonne etichettate

1. Rappresentare i gruppi sanguigni in una tabella delle frequenze (assolute)
2. Rappresentare i gruppi sanguigni in una tabella delle frequenze relative
3. Restituire i nomi ed i cognomi delle persone che hanno gruppo sanguigno B
4. Restituire i cognomi delle persone che hanno gruppo sanguigno B e che si chiamano Cristina

In [5]:
campnomi <- sample(totnomi, 100, replace=TRUE)
campcognomi <- sample(totcognomi, 100, replace=TRUE)
campgruppi <- sample(totgruppi, 100, replace=TRUE)
donatori <- data.frame(nomi = campnomi, cognomi = campcognomi, gruppi = campgruppi)

In [9]:
grassfreq <- table(donatori[["gruppi"]])
grassfreq


 0  A AB  B 
29 26 31 14 

In [11]:
grrelfreq <- grassfreq/length(donatori[[1]])
grrelfreq


   0    A   AB    B 
0.29 0.26 0.31 0.14 

In [12]:
donatori[donatori$gruppi=="B", c("nomi", "cognomi")]

nomi      cognomi  
3  Tommaso   Marino   
9  Cristina  Conti    
19 Luca      Esposito 
32 Alessia   Verdi    
34 Francesca Neri     
48 Alessia   Brambilla
57 Alessia   Rossi    
60 Alessia   Verdi    
64 Mario     Bianchi  
66 Luca      Colombo  
82 Pino      Bianchi  
86 Pino      Greco    
90 Mario     Esposito 
92 Marco     Ferrari

In [14]:
donatori[donatori$gruppi=="B" & donatori$nomi=="Cristina", "cognomi"]

[1] Conti
14 Levels: Bianchi Brambilla Colombo Conti Esposito Ferrari Gallo ... Verdi

# Esercizio 2

1. Scaricare il dataset mtcars.csv
2. Importare il dataset creando un dataframe
3. Visualizzare la struttura del dataset
4. Visualizzare le prime 6 righe del dataset
5. Convertire i pesi (wt) da libbre a chilogrammi e sovrascrivere il dataframe (attenzione i valori iniziali sono 1000 lb)
6. Stampare la riga dell’auto più pesante
7. Calcolare la tabella delle frequenze assolute del numero dei cilindri (cyl)
8. Determinare le auto che hanno 4 cilindri e più di 70 cavalli (hp) e stamparne modello (model) e peso (wt)(provate sia con subset che con [])

In [33]:
mtcars <- read.csv(file = "mtcars.csv")
str(mtcars)

'data.frame':	32 obs. of  12 variables:
 $ model: Factor w/ 32 levels "AMC Javelin",..: 18 19 5 13 14 31 7 21 20 22 ...
 $ mpg  : num  21 21 22.8 21.4 18.7 18.1 14.3 24.4 22.8 19.2 ...
 $ cyl  : int  6 6 NA 6 8 6 8 4 4 6 ...
 $ disp : num  160 160 108 258 360 ...
 $ hp   : int  110 110 93 110 175 105 245 62 NA 123 ...
 $ drat : num  3.9 3.9 3.85 3.08 3.15 2.76 3.21 3.69 3.92 3.92 ...
 $ wt   : num  2.62 2.88 2.32 3.21 3.44 ...
 $ qsec : num  16.5 17 18.6 19.4 17 ...
 $ vs   : int  0 0 1 1 0 1 0 1 1 1 ...
 $ am   : int  1 1 1 0 0 0 0 0 0 0 ...
 $ gear : int  4 4 4 3 3 3 3 4 4 4 ...
 $ carb : int  4 4 1 1 2 1 4 2 2 4 ...


In [31]:
mtcars[1:6, ]

model             mpg  cyl disp hp  drat wt    qsec  vs am gear carb
1 Mazda RX4         21.0  6  160  110 3.90 2.620 16.46 0  1  4    4   
2 Mazda RX4 Wag     21.0  6  160  110 3.90 2.875 17.02 0  1  4    4   
3 Datsun 710        22.8 NA  108   93 3.85 2.320 18.61 1  1  4    1   
4 Hornet 4 Drive    21.4  6  258  110 3.08 3.215 19.44 1  0  3    1   
5 Hornet Sportabout 18.7  8  360  175 3.15 3.440 17.02 0  0  3    2   
6 Valiant           18.1  6  225  105 2.76 3.460 20.22 1  0  3    1

In [34]:
mtcars[["wt"]] <- mtcars[["wt"]]/0.0022046
mtcars[1:6, ]

model             mpg  cyl disp hp  drat wt       qsec  vs am gear carb
1 Mazda RX4         21.0  6  160  110 3.90 1188.424 16.46 0  1  4    4   
2 Mazda RX4 Wag     21.0  6  160  110 3.90 1304.091 17.02 0  1  4    4   
3 Datsun 710        22.8 NA  108   93 3.85 1052.345 18.61 1  1  4    1   
4 Hornet 4 Drive    21.4  6  258  110 3.08 1458.314 19.44 1  0  3    1   
5 Hornet Sportabout 18.7  8  360  175 3.15 1560.374 17.02 0  0  3    2   
6 Valiant           18.1  6  225  105 2.76 1569.446 20.22 1  0  3    1

In [35]:
mtcars[mtcars[["wt"]]==max(mtcars[["wt"]]), ]

model               mpg  cyl disp hp  drat wt      qsec  vs am gear carb
16 Lincoln Continental 10.4 8   460  215 3    2460.31 17.82 0  0  3    4

In [36]:
table(mtcars[["cyl"]])


 4  6  8 
10  7 14 

In [43]:
mtcars[mtcars[["cyl"]]==4 & mtcars[["hp"]]>70 & !is.na(mtcars[["cyl"]]) & !is.na(mtcars[["hp"]]), c("model", "wt")]

model         wt       
21 Toyota Corona 1118.1167
27 Porsche 914-2  970.6976
28 Lotus Europa   686.2923
32 Volvo 142E    1260.9997

In [45]:
subset(mtcars, cyl==4 & hp>70 & !is.na(cyl) & !is.na(hp))[c("model", "wt")]

model         wt       
21 Toyota Corona 1118.1167
27 Porsche 914-2  970.6976
28 Lotus Europa   686.2923
32 Volvo 142E    1260.9997

# Esercizio 3

1. Scaricare il dataset atleti.csv
2. Importare il dataset creando un dataframe e controllarne la struttura
3. I dati dei pesi soddisfano le tre condizioni necessarie affinché siano approsimativamente normali? Ross pag. 105
4. I dati dell’età sono simmetrici intorno al valore medio? Ross pag 56

In [48]:
atleti <- read.csv(file = "atleti.csv")
str(atleti)

'data.frame':	1000 obs. of  3 variables:
 $ peso   : int  69 70 72 68 72 72 64 70 67 65 ...
 $ altezza: num  164 173 176 175 171 ...
 $ eta    : int  14 11 28 15 26 31 23 58 16 29 ...


In [53]:
meanpeso <- mean(atleti$peso)
sdpeso <- sd(atleti$peso)
sum(atleti$peso>=(meanpeso-sdpeso) & atleti$peso<=(meanpeso+sdpeso))/length(atleti$peso)
sum(atleti$peso>=(meanpeso-2*sdpeso) & atleti$peso<=(meanpeso+2*sdpeso))/length(atleti$peso)
sum(atleti$peso>=(meanpeso-3*sdpeso) & atleti$peso<=(meanpeso+3*sdpeso))/length(atleti$peso)

[1] 0.68

[1] 0.966

[1] 1

In [71]:
meaneta <- round(mean(atleti$eta), digits=0)
simmfreq <- matrix(c(sum(atleti$eta==meaneta), sum(atleti$eta==meaneta)), nrow=1)
for(c in 1:50) {
    simmfreq <- rbind(simmfreq, c(sum(atleti$eta==meaneta-c), sum(atleti$eta==meaneta+c)))
}
sum(simmfreq[, 1]==simmfreq[, 2])==nrow(simmfreq)

[1] FALSE